File: bacha/bacha.2022.03.14/3.kt
import kotlin.math.*
fun main() {
    val n = readLine()!!.toInt()
    var ans = 2000000000
    for (i in 1 .. n) {
        val (a, p, x) = readLine()!!.split(" ").map{ it.toInt() }
        if (x - a > 0) ans = min(ans, p)
    }
    println(
        if (ans == 2000000000) -1
        else ans
           )
}

File: abs/3.py
n = int(input())
a = list(map(int, input().split()))

flag = True
ans = 0
while flag:
	for i in range(n):
		if a[i] % 2 == 0:
			a[i] //= 2
		else:
			flag = False
	ans += 1
ans -= 1
print(ans)
